manta-pay/concrete-protocol.tex

```diff
@@ -1,49 +1,65 @@
-    For $\KDF$, we use Blake2s~\cite{rfc7693} with magic-number salt.
-\end{definitiontoc}
-
-\begin{definitiontoc}{Dynamic Cryptographic Accumulator}
+\begin{definition}[Commitment Schemes]
+    The protocol features two different commitment schemes: $\COM^{\UTXO}$ the \UTXO{} Commitment Scheme and $\COM^{\VN}$ the Void Number Commitment Scheme. Both commitment schemes use \Poseidon{} as the underlying cryptographic primitive. The \UTXO{} uses an arity-5 \Poseidon{} with the following mapping:
+    \[\COM^{\UTXO}(r, \pk, \asset) := \Poseidon_5(d, r, \X(\pk), \Y(\pk), \asset.\ID, \asset.\VALUE)\]
+    where $d = \HashToScalar(\domainstr{manta-pay/1.0.0/com-utxo})$ the domain separation element.
+    For the Void Number Commitment Scheme we use an arity-3 \Poseidon{} with the following mapping:
+    \[\COM^{\VN}(\ak, h) := \Poseidon_3(d, \X(\ak), \Y(\ak), h)\]
+    where $d = \HashToScalar(\domainstr{manta-pay/1.0.0/com-vn})$ the domain separation element.
+\end{definition}
+
+\begin{definition}[Hash Functions]
+    The protocol features two additional hash functions: $\KDF^{\vk}$ the viewing key derivation function and $H^{\UTXO}$ the \UTXO{} hash function. Both hash functions use \Poseidon{} as the underlying cryptographic primitive. The viewing key derivation function uses an arity-2 \Poseidon{}
+    \[\KDF^{\vk}(\ak) := \Poseidon_2(d, \X(\ak), \Y(\ak))\]
+    where $d = \HashToScalar(\domainstr{manta-pay/1.0.0/kdf-vk})$ the domain separation element. Since the target type of this hash function is the scalar field $\mathbb{S}$ instead of $\F$, we reduce the result modulo the order of $\mathbb{S}$.
+    For the \UTXO{} hash function we use an arity-4 \Poseidon{}
+    \[H^{\UTXO}(t, \pa, \cm) := \Poseidon_4(d, t, \pa.\ID, \pa.\VALUE, \cm)\]
+    where $d = \HashToScalar(\domainstr{manta-pay/1.0.0/utxo-hash})$ the domain separation element.
+\end{definition}
+
+\begin{definition}[Signature Scheme]
+    For the signature scheme we use Schnorr signature over the inner curve, $\mathbb{G}$.
+\end{definition}
+
+\begin{definition}[Authenticated Encryption Scheme]
+    For $\AUTHin$ and $\AUTHout$ we use the $\Poseidon$ permutation as the permutation of a duplex sponge encryption protocol.
+\end{definition}
+
+\begin{definition}[Dynamic Cryptographic Accumulator]
     For $\DCA$, we use a Merkle Tree with $\Poseidon_2$ as the inner node combining hash function and no leaf hash function. It is safe to omit the leaf hash function in this case because the leaf values are already the outputs of a hash function and cannot be directly controlled.
-\end{definitiontoc}
+\end{definition}
 
-\begin{definitiontoc}{Non-Interactive Zero-Knowledge Proving System} \label{def:concrete-zkp}
+\begin{definition}[Non-Interactive Zero-Knowledge Proving System] \label{def:concrete-zkp}
     For $\NIZK$, the protocol can use any non-interactive zero-knowledge proving system like Groth16 \cite{KRRS21USENIX} and/or PLONK/PLONKUP~\cite{plonk,plonkup}. 
```

Comment:
Are we though? As far as I know, we're only using Groth16 even if MantaPay supports other systems. Also, refer to the trusted setup docs for key generation.